### PR TITLE
Add unit testing framework and fix bug found.

### PR DESCRIPTION
**Pull request by Hoikas**
A long standing issue has been that when a `STATEDESC` is updated, the values are lost. So, I wanted to add tests for updating SDL blobs. There was a rather dire comment about there not being a testing framework, so this adds one based on Catch2. Note that Catch2 2.x was used instead of Catch2 3.x because the 3.x adds a static library component, which I found undesirable. After adding the basic SDL tests, I found that the SDL varmap is not being initialized in all situations.

### CODE REVIEW

File: Tests/Test_SDL.cpp [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+            return false;
+        }
+        fwrite(s_SdlDescriptor.c_str(), sizeof(char), s_SdlDescriptor.size(), f);
+        fclose(f);
+
+        SDL::DescriptorDb::LoadDescriptors(tempDir);
+        unlink(sdlFilePath.c_str());
+        rmdir(tempDirResult);
+        return true;
+    } while (0);
```
Comment (zrax) [inline Tests/Test_SDL.cpp:85]: I don't see any `break` or `continue`, and this is the only scope in the function...  Why is the `do ... while` necessary?

Comment (Hoikas) [reply Tests/Test_SDL.cpp:88]: Oof. I had the thought of using that construct for resource management and promptly failed to manage my resources correctly. 🤦‍♂️ Fixed now.

File: Tests/Test_SDL.cpp [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+static bool LoadDescriptors()
+{
+    static bool result = false;
+    if (result)
+        return true;
+
+    ST::string sdlFilePath;
+    char tempDir[256]{};
+    const char kTemplate[] = "DirtSandSDLTestXXXXXX";
+    strcpy(tempDir, kTemplate);
```
Comment (zrax) [inline Tests/Test_SDL.cpp:68]:
```suggestion
    char tempDir[256] = "/tmp/DirtSandSDLTestXXXXXX";
```

File: Tests/Test_SDL.cpp [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+    do {
+        char* tempDirResult = mkdtemp(tempDir);
+        if (!tempDirResult) {
+            fprintf(stderr, "Failed to create a temporary directory for SDL.\n");
+            break;
+        }
+
+        sdlFilePath = ST::format("{}/Test.sdl", tempDirResult);
+        fprintf(stderr, "%s\n%s\n%s\n", sdlFilePath.c_str(), tempDir, tempDirResult);
```
Comment (zrax) [inline Tests/Test_SDL.cpp:77]: Debug line left in?

File: Tests/Test_SDL.cpp [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+        fwrite(s_SdlDescriptor.c_str(), sizeof(char), s_SdlDescriptor.size(), f);
+        fclose(f);
+
+        SDL::DescriptorDb::LoadDescriptors(tempDirResult);
+        result = true;
+    } while (0);
+
+    if (!sdlFilePath.empty())
+        unlink(sdlFilePath.c_str());
+    if (*tempDir)
```
Comment (zrax) [inline Tests/Test_SDL.cpp:92]: This test should be against `tempDirResult` -- `mkdtemp` doesn't clear the template upon failure, as far as I can tell.

File: Tests/Test_SDL.cpp [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+
+    auto boolVarIdx = desc->m_varmap.find("bTestVar1");
+    auto intVarIdx = desc->m_varmap.find("iTestVar3");
+
+    REQUIRE(boolVarIdx != desc->m_varmap.end());
+    REQUIRE(intVarIdx != desc->m_varmap.end());
+
+    SDL::State state(desc);
+    state.data()->m_vars[boolVarIdx->second].data()->m_bool = new bool[1] { true };
+    state.data()->m_vars[intVarIdx->second].data()->m_int = new int32_t[1] { 6 };
```
Comment (zrax) [inline Tests/Test_SDL.cpp:175]: These are leaks as written (the storage is already allocated):

```suggestion
    state.data()->m_vars[boolVarIdx->second].data()->m_bool[0] = true;
    state.data()->m_vars[intVarIdx->second].data()->m_int[0] = 6;
```

**Comment by Hoikas** [comment]
I think I've address the most recent review comments.

File: Tests/Test_SDL.cpp [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+        fclose(f);
+
+        SDL::DescriptorDb::LoadDescriptors(tempDirResult);
+        result = true;
+    } while (0);
+
+    if (!sdlFilePath.empty())
+        unlink(sdlFilePath.c_str());
+    if (tempDir)
+        rmdir(tempDir);
```
Comment (zrax) [inline Tests/Test_SDL.cpp:91]:
```suggestion
    if (tempDirResult)
        rmdir(tempDirResult);
```

**Review by zrax — APPROVED** [review]

**Pull request merged**